### PR TITLE
Disable redux for GENX target

### PR DESCRIPTION
**Pull request by quintinwang5**
Fix https://github.com/intel/intel-xpu-backend-for-triton/issues/412 model: `AllenaiLongformerBase` by disabling `nvvm.redux.sync` emit.
```
failed to legalize operation 'tt.reduce' that was explicitly marked illegal
```

### CODE REVIEW

File: lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp [hunk 1]
```diff
@@ -212,39 +212,44 @@ struct ReduceOpConversion
...
-        for (unsigned i = 0; i < acc.size(); ++i) {
-          unsigned bitwidth = acc[i].getType().cast<IntegerType>().getWidth();
-          if (bitwidth < 32) {
-            if (*kind == NVVM::ReduxKind::MIN || *kind == NVVM::ReduxKind::MAX)
-              acc[i] = sext(i32_ty, acc[i]);
-            else
-              acc[i] = zext(i32_ty, acc[i]);
+                  unsigned numLaneToReduce, unsigned interleave,
+                  Target target) const {
+    if (target != Target::GENX) {
```
Comment (whitneywhtsang) [inline lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp:217]:
```suggestion
    if (target == Target::GENX)
      return;
```

Comment (quintinwang5) [reply lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp:217]: @whitneywhtsang Cannot return here. We need to use the path below. https://github.com/intel/intel-xpu-backend-for-triton/blob/6f87de1024cb28e27995259033106f26c03f4823/lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp#L251

**Review by etiotto — COMMENTED** [review]
LGTM -thanks!

**Comment by pengtu** [comment]
@quintinwang5 @whitneywhtsang @etiotto:  We need to have an issue to track adding WarpReduce to XPU using SPIRV OpGroupReduce operations, for example: https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpGroupFMin

**Comment by quintinwang5** [comment]
> @quintinwang5 @whitneywhtsang @etiotto: We need to have an issue to track adding WarpReduce to XPU using SPIRV OpGroupReduce operations, for example: https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpGroupFMin

I think it's better to add WarpReduce to GENX, as WarpShuffle. IGC has these intrinsics.

**Comment by pengtu** [comment]
> > @quintinwang5 @whitneywhtsang @etiotto: We need to have an issue to track adding WarpReduce to XPU using SPIRV OpGroupReduce operations, for example: https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpGroupFMin
> 
> I think it's better to add WarpReduce to GENX, as WarpShuffle. IGC has these intrinsics.

Good idea to add it to GENX!  IGC team prefers us to use SPIRV constructs if available.  Please file an issue so we can implement it.

**Comment by etiotto** [comment]
> > > @quintinwang5 @whitneywhtsang @etiotto: We need to have an issue to track adding WarpReduce to XPU using SPIRV OpGroupReduce operations, for example: https://registry.khronos.org/SPIR-V/specs/unified1/SPIRV.html#OpGroupFMin
> > 
> > 
> > I think it's better to add WarpReduce to GENX, as WarpShuffle. IGC has these intrinsics.
> 
> Good idea to add it to GENX! IGC team prefers us to use SPIRV constructs if available. Please file an issue so we can implement it.

I filed this isse agains the GEN dialect: https://github.com/intel/llvm/issues/12609

**Pull request merged**